### DISCUSSION TITLE
CoupledVarThesholdNodalBCModifier type system exists?

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#22172] [Q&A General] [answered]
I have a mesh with values at the nodes of 0 or 1. If the indicator value is 1, I want to apply a DirichletBC to that node and for the rest of the nodes, I just want to solve some other physics. The indicator field will change every timestep, and can't be determined beforehand. Can someone point me to some code that shows the changing of node sets based on a value or something that could perform the action above? Thank you!

**Comment by hugary1995** [comment, marked as answer]
You can follow the element subdomain modifier code. IIRC modifying the nodeset was actually the easy part of all. You'd modify the boundary info and call meshChanged to reinit the equation system.

**Reply by GiudGiud** [reply]
I thought there was some nodeset modifying code in a PR somewhere. I’ll check when I get back

**Reply by maxnezdyur** [reply]
Let me know if you find it. I created my own that works for my application but will probably need some changes to be general purpose enough to add into MOOSE.

**Reply by GiudGiud** [reply]
Looking at recent activity on this, https://github.com/idaholab/moose/issues/21579, seems we do not have this

We're definitely interested if you have something working
